### PR TITLE
Fix bugs in the library and required cleanups on the sample app

### LINKED ISSUE #212: Bugs encountered when using Sample app
- [x] MapActivity crashing if not openned with points dropped passed
- [x] Offline map downloading service throwing exception if provided an `http` or `https` url as the styleUrl - This should be allowed as long as it's usable by Mapbox

### LINKED ISSUE #213: Sample app bug cleanup
- [x] On openning most of the screens/activities from the Navigation drawer, the title of the activity remains as Kujaku Sample
- [x] Automatically version sample app based on latest tag information from Github

This enables us to easily track changes and track source of errors that are reported on Crashlytics when developers are working on features using the sample app

### PR DESCRIPTION
**Pull request by ekigamba**
Fixes #212 
Fixes #213

### CODE REVIEW

File: sample/build.gradle [hunk 1]
```diff
@@ -24,8 +74,8 @@ android {
         applicationId "io.ona.kujaku.sample"
         minSdkVersion 16
         targetSdkVersion rootProject.ext.targetSdkVersion
-        versionCode 1
-        versionName "1.0"
+        versionCode getMasterCommitCount()
```
Comment (vincent-karuri) [inline sample/build.gradle]: How would one test if this works?

File: sample/build.gradle [hunk 2]
```diff
@@ -24,8 +74,8 @@ android {
...
+        versionName getVersionName()
```
Comment (vincent-karuri) [inline sample/build.gradle]: How would one test if this works?

Comment (ekigamba) [reply sample/build.gradle]: Run the app on your device and check the version name, it should be somewhere around `v0.1.4` and some other identifiers

File: sample/build.gradle (hunk 1 above)
Comment (ekigamba) [reply sample/build.gradle]: Merge the code to your local master, make sure to build the app & check your sample module `Android.manifest` file

Add more commits on top of your master and check it again, the version code should change due to the extra commits

**Pull request merged**